backend/de.metas.swat/de.metas.swat.base/src/test/java/de/metas/order/createFrom/po_from_so/impl/CreatePOFromSOsAggregationKeyBuilderTest.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * #%L
+ * de.metas.swat.base
+ * %%
+ * Copyright (C) 2021 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.order.createFrom.po_from_so.impl;
+
+import de.metas.adempiere.model.I_M_Product;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.model.PlainContextAware;
+import org.adempiere.test.AdempiereTestHelper;
+import org.compiere.model.I_AD_SysConfig;
+import org.compiere.model.I_C_BPartner;
+import org.compiere.model.I_C_BPartner_Product;
+import org.compiere.model.I_C_Order;
+import org.compiere.model.I_C_OrderLine;
+import org.compiere.util.Env;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static de.metas.order.createFrom.po_from_so.impl.CreatePOFromSOsAggregationKeyBuilder.KEY_SKIP;
+import static de.metas.order.createFrom.po_from_so.impl.CreatePOFromSOsAggregationKeyBuilder.SYSCONFIG_ON_MISSING_C_B_PARTNER_PRODUCT;
+import static de.metas.order.createFrom.po_from_so.impl.CreatePOFromSOsAggregationKeyBuilder.ON_MISSING_C_B_PARTNER_PRODUCT_ERROR;
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
+import static org.assertj.core.api.Assertions.*;
+
+public class CreatePOFromSOsAggregationKeyBuilderTest
+{
+	private CreatePOFromSOsAggregationKeyBuilder createPOFromSOsAggregationKeyBuilder;
+	private I_M_Product productRecord;
+	private I_C_BPartner customerBPartner;
+	private I_C_Order salesOrderRecord;
+
+	@BeforeEach
+	void init()
+	{
+		AdempiereTestHelper.get().init();
+
+		productRecord = newInstance(I_M_Product.class);
```

Comment:
ok, but note for next time you can also use `de.metas.business.BusinessTestHelper` to create such test data